### PR TITLE
Features/jump to error states #120234287

### PR DESCRIPTION
**Pull request by kaptron**

### CODE REVIEW

File: app/assets/javascripts/short-form/ShortFormApplicationController.js.coffee [hunk 1]
```diff
@@ -115,7 +116,14 @@ ShortFormApplicationController = (
...
     else
-      $scope.hideAlert = false
+      $scope.handleErrorState()
+
+  $scope.handleErrorState = ->
+    # show error alert
+    $scope.hideAlert = false
+    el = angular.element(document.getElementById('short-form-wrapper'))
+    # uses duScroll aka 'angular-scroll' module
+    $document.scrollToElement(el, 0, 400)
```
Comment (chrisdolendo) [inline app/assets/javascripts/short-form/ShortFormApplicationController.js.coffee]: what is the significance of 0 and 400?

File: app/assets/javascripts/short-form/directives/checkboxPreferenceItemDirective.js.coffee [hunk 2]
```diff
@@ -25,4 +25,8 @@ angular.module('dahlia.directives')
         scope.householdMembers
 
     scope.only_applicant_eligible = () ->
-      (scope.eligible_members().length == 1) && (scope.eligible_members()[0] == scope.applicant)
+      applicant_only = (scope.eligible_members().length == 1) && (scope.eligible_members()[0] == scope.applicant)
+      if applicant_only
+        # even though the form input is hidden we automatically set the value to the applicant
```
Comment (chrisdolendo) [inline app/assets/javascripts/short-form/directives/checkboxPreferenceItemDirective.js.coffee]: thanks for catching this!

**Comment by chrisdolendo** [comment]
coolio! :+1:

File: app/assets/javascripts/short-form/ShortFormApplicationController.js.coffee [hunk 3]
```diff
@@ -115,7 +116,16 @@ ShortFormApplicationController = (
...
+      $scope.handleErrorState()
+
+  $scope.handleErrorState = ->
+    # show error alert
+    $scope.hideAlert = false
+    el = angular.element(document.getElementById('short-form-wrapper'))
+    # uses duScroll aka 'angular-scroll' module
+    topOffset = 0
+    duration = 400 # animation speed in ms
+    $document.scrollToElement(el, topOffset, duration)
```
Comment (kaptron) [inline app/assets/javascripts/short-form/ShortFormApplicationController.js.coffee]: @chrisdolendo good call on the `0, 400` not having any context for other devs viewing this code, I updated that to be more clear

Comment (chrisdolendo) [inline app/assets/javascripts/short-form/ShortFormApplicationController.js.coffee]: Awesome! looks cool to me!

**Pull request merged**